data_model/derive/src/lib.rs

```diff
@@ -1,8 +1,80 @@
-//! Crate with `Filter` and `EventFilter` derive macro
-#![allow(clippy::expect_used, clippy::panic, clippy::too_many_lines)]
+//! A crate containing various derive macros for `data_model`
 
 use proc_macro::TokenStream;
-use quote::{format_ident, quote};
+use syn::parse_macro_input;
+
+mod filter;
+mod id;
+
+/// A derive macro for `Identifiable` trait and id-based comparison traits. Currently supports derivations only for
+/// `IdBox`, `Event` enums, and structs from the `data_model` crate that don't have generic parameters.
+///
+/// As such, the macro introduces a new
+/// outer attribute `id` for the entities it is derived from. This attribute should
+/// be supplied with the associated type to be used in `impl Identifiable`. The type
+/// should be supplied as a string literal that constitutes a
+/// legal Rust type path.
+///
+/// As this macro also derives an implementation of `Ord`, `PartialOrd`, `Eq`, `PartialEq` and `Hash` traits that always
+/// conforms to the same implementation principles based on ids of the entities.
+/// Thus none of the entities that derive this macro should derive neither of the aforementioned traits,
+/// as they will be overridden.
+///
+/// Another warning: additional care should be exercised when using this macro for types that could be connected with
+/// anything concerning the trigger `Set` structure, or anywhere else where the specifics of `Ord`
+/// implementation become important, as subtle bugs may appear.
```

Comment:
I don't like seeing this comment especially since you've opened the issue.
May I ask what's the big problem here with `TriggerSet`?

Reply:
Added this as requested by @Arjentix in [this comment](https://github.com/hyperledger/iroha/pull/2476#discussion_r920046366). Might not be relevant anymore.

Reply:
I see. I think it should be the other way around. There is a contract this derive and `Identifiable` offer and that is uniqueness of `Id`. If your structure can't be uniquely identifiable it should not in any circumstance implement `Identifiable` or this derive. That's the rule, there are no exceptions

Reply:
Changed the doc to reflect that.